### DISCUSSION TITLE
Parsing .map files

### DISCUSSION THREAD

**Discussion by SelectiveHoe** [OpenSAGE/OpenSAGE#582] [Q&A]
Hi, i working on fun made remake Generals Zero Hour and i faced a problem.

I want to parse .map files to generate a map in Unity using maps from game and World Builder as redactor for it, but i can't find information about .map files. Is they have structure wich i can parse and use it in C# code or somthing like that?

May be i need to unzip or decrypt first but i don't know how. I will be pleased any help.

**Comment by Skleni** [comment]
You can start by taking a look at the [MapFile](https://github.com/OpenSAGE/OpenSAGE/blob/master/src/OpenSage.Game/Data/Map/MapFile.cs) class.
